recipes/full_finetune_distributed.py

```diff
@@ -314,10 +314,19 @@ def setup(self, cfg: DictConfig) -> None:
         self._compile_model = compile_bool
         self._compile_loss = compile_bool
         self._compile_optimizer_step = compile_bool
+        self._compile_scale_grads = compile_bool
         if isinstance(compile, DictConfig):
             self._compile_model = compile.get("model", True)
             self._compile_loss = compile.get("loss", True)
             self._compile_optimizer_step = compile.get("optimizer_step", False)
+            self._compile_scale_grads = compile.get("scale_grads", True)
```

Comment:
nit but I actually wonder whether it's worth having a separate flag for this. I think model, loss, optimizer step are all pretty clear but this one may be a bit niche for someone to look at and immediately understand (and also I think it's unlikely for someone to actually want to experiment with). Could even just use some heuristic like `self._compile_scale_grads = all([self._compile_model, self._compile_loss, self._compile_optimizer_step])`